### PR TITLE
add best practice try/catch, timeout, specifically with extended test…

### PR DESCRIPTION
**Pull request by gabemontero**
…s in mind

the diff came out a little wonky - try viewing with ignore white space option

but in general existing contents were put inside 
```
try {
   timeout(...) {
   } 
} catch ... {
}
```

@openshift/devex

### CODE REVIEW

**Comment by gabemontero** [comment]
[testextended][extended:core(openshift pipeline build)]

**Comment by gabemontero** [comment]
fyi ... adding `?w=1` to the file diff url is the ignore whitespace trigger ... the file diff renders as I described when employed

File: examples/jenkins/pipeline/bluegreen-pipeline.yaml [hunk 1]
```diff
@@ -26,43 +26,52 @@ objects:
...
-            stage("Test") {
-              input message: "Test deployment: http://${routeHost}. Approve?", id: "approval"
-            }
-
-            stage("Go Live") {
-              sh "oc set -n ${project} route-backends ${appName} ${appName}-${tag}=100 ${appName}-${altTag}=0 --loglevel=4"
-            }
-          }
+          try {
+             timeout(time: 2, unit: 'HOURS') {
```
Comment (bparees) [inline examples/jenkins/pipeline/bluegreen-pipeline.yaml]: 2 hours?  i'm worried about making our test runs take a really long time to fail.  is there something in the extended test itself that's going to cap this?  regardless i'd hope we could bring it down to 15-20 minutes tops.

Comment (gabemontero) [inline examples/jenkins/pipeline/bluegreen-pipeline.yaml]: yeah 2 hours was *NOT* based on any examination of deployment times or some such ... just a "make it really big so it is not a factor" rationale

I'll switch to 20 minutes.

**Comment by gabemontero** [comment]
ah ... still waiting on the new client plugin version with the `exists` since we still can't update the centos image ... will address in separate PR

the tests based on this PR's changes passed, but will clean up above, plus reduce  timeout

**Comment by gabemontero** [comment]
ok updates pushed

**Comment by bparees** [comment]
lgtm if it passes tests

**Comment by gabemontero** [comment]
and they just passed :-)

**Comment by bparees** [comment]
[merge][severity:bug]

**Comment by bparees** [comment]
looks like the passing run was your previous commit.  this run failed with 
```
+ oc start-build openshift-jee-sample-docker --from-file=target/ROOT.war --follow -n extended-test-jenkins-pipeline-37mv8-n85lx
Uploading file "target/ROOT.war" as binary input for the build ...
Unable to connect to the server: net/http: TLS handshake timeout
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // timeout
[Pipeline] echo
in catch block
[Pipeline] echo
Caught: hudson.AbortException: script returned exit code 1
```

does this imply the 20 minute timeout might not be long enough?

**Comment by gabemontero** [comment]
the extended failure this time was the very familiar flake https://github.com/openshift/origin/issues/13984

Further confirmation that the `--follow` results in the TLS handshake timeout, while in debug PR https://github.com/openshift/origin/pull/14013, where that has been replaced with `--wait`, we get the more generic `exit status -1`

And I really am beginning to wonder if there is any correlation with https://github.com/openshift/origin/issues/14148 and some of the console's issues with follow since the rebase.

**Comment by gabemontero** [comment]
On Thu, May 25, 2017 at 1:26 PM, Ben Parees <notifications@github.com>
wrote:

> looks like the passing run was your previous commit. this run failed with
>
> + oc start-build openshift-jee-sample-docker --from-file=target/ROOT.war --follow -n extended-test-jenkins-pipeline-37mv8-n85lx
> Uploading file "target/ROOT.war" as binary input for the build ...
> Unable to connect to the server: net/http: TLS handshake timeout
> [Pipeline] }
> [Pipeline] // stage
> [Pipeline] }
> [Pipeline] // node
> [Pipeline] }
> [Pipeline] // timeout
> [Pipeline] echo
> in catch block
> [Pipeline] echo
> Caught: hudson.AbortException: script returned exit code 1
>
> does this imply the 20 minute timeout might not be long enough?
>
Short answer: no

See https://github.com/openshift/origin/pull/14354#issuecomment-304071078

Also, in looking at the various timestamps, 20 minutes had not expired.

**Comment by bparees** [comment]
[merge][severity:bug]

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/13980
flake https://github.com/openshift/origin/issues/10773
flake https://github.com/openshift/origin/issues/14385
[merge][severity:bug]

**Pull request merged**